src/titiler/xarray_api/factory.py

```diff
@@ -0,0 +1,196 @@
+                buckets = list(
+                    zip(values, [values[i + 1] for i in range(len(values) - 1)])
+                )
+                hist_dict = []
+                for idx, bucket in enumerate(buckets):
+                    hist_dict.append({"bucket": bucket, "value": counts[idx]})
+                return hist_dict
+
+    def map_viewer(self) -> None:
+        """Register /map endpoints"""
+
+        @self.router.get("/{tileMatrixSetId}/map", response_class=HTMLResponse)
+        def map_viewer(
+            request: Request,
+            tileMatrixSetId: Annotated[  # type: ignore
+                Literal[tuple(self.supported_tms.list())],
+                "Identifier selecting one of the supported TileMatrixSetIds",
+            ],
+            url: Annotated[Optional[str], Query(description="Dataset URL")] = None,
+            variable: Annotated[
+                Optional[str],
+                Query(description="Xarray Variable"),
+            ] = None,
+            group: Annotated[
+                Optional[int],
+                Query(
+                    description="Select a specific zarr group from a zarr hierarchy, can be for pyramids or datasets. Can be used to open a dataset in HDF5 files."
+                ),
+            ] = None,
+            decode_times: Annotated[
+                bool,
+                Query(
+                    title="decode_times",
+                    description="Whether to decode times",
+                ),
+            ] = True,
+            drop_dim: Annotated[
+                Optional[str],
+                Query(description="Dimension to drop"),
+            ] = None,
+            datetime: Annotated[
+                Optional[str], Query(description="Slice of time to read (if available)")
+            ] = None,
+            tile_format: Annotated[
+                Optional[ImageType],
+                Query(
+                    description="Default will be automatically defined if the output image needs a mask (png) or not (jpeg).",
+                ),
+            ] = None,
+            tile_scale: Annotated[
+                int,
+                Query(
+                    gt=0, lt=4, description="Tile size scale. 1=256x256, 2=512x512..."
+                ),
+            ] = 1,
+            minzoom: Annotated[
+                Optional[int],
+                Query(description="Overwrite default minzoom."),
+            ] = None,
+            maxzoom: Annotated[
+                Optional[int],
+                Query(description="Overwrite default maxzoom."),
+            ] = None,
+            post_process=Depends(self.process_dependency),
+            rescale=Depends(self.rescale_dependency),
+            color_formula=Depends(ColorFormulaParams),
+            colormap=Depends(self.colormap_dependency),
+            render_params=Depends(self.render_dependency),
+            nodata=Depends(nodata_dependency),
```

Comment:
nodata should be defined by the `dataset_dependency` https://github.com/developmentseed/titiler/blob/main/src/titiler/xarray/titiler/xarray/factory.py#L44